addon/mixins/legend.js

```diff
@@ -91,32 +98,44 @@ export default Ember.Mixin.create({
     };
   }),
 
-  // Place each legend item, breaking across rows. Center them if there is one
-  // row
-  // Ideally legend items would be centered to the very middle of the graph,
-  // this is made difficult by the fact that we want the icons to line up in
-  // nice columns and that the labels are variable length
-  legendItemAttrs: Ember.computed('legendItemWidth', 'legendItemHeight', 'numLegendItemsPerRow',
-     'legendItems.length', 'numLegendRows', 'averageLegendLabelWidth', function() {
-
+  // Place each legend item such that the legend rows appear centered to the graph.
+  // Spacing between legend items must be constant and equal to 2*legendLabelPadding = 20px.
+  // This breaks the old behavior where legend icons are placed in nice columns but caused
+  // in-efficient use of space for legend text. Legend icons will not be in nicely aligned 
+  // in the columns anymore to allow more space for variable sized labels.
+  legendItemAttrs: Ember.computed('legendItemWidth', 'legendItemHeight', 'legendIconRadius',
+     'legendLabelPadding', 'legendRowWidths', 'numLegendItemsByRows', function() {
 
+    var legendRowWidths = this.get('legendRowWidths');
     var legendItemWidth = this.get('legendItemWidth');
     var legendItemHeight = this.get('legendItemHeight');
-    var numItemsPerRow = this.get('numLegendItemsPerRow');
-    var numAllItems = this.get('legendItems.length');
-    var isSingleRow = this.get('numLegendRows') === 1;
+    var legendLabelPadding = this.get('legendLabelPadding');
+    var legendIconRadius = this.get('legendIconRadius');
+    var numLegendItemsByRows = this.get('numLegendItemsByRows');
 
-    var _this = this;
+    // Track the space already alloted to a legend.
+    // This is used to translate the next legend in the row.
+    var usedWidth = 0;
     return {
       "class": 'legend-item',
       width: legendItemWidth,
       'stroke-width': 0,
-      transform: function(d, i) {
-        var col = i % numItemsPerRow;
-        var row = Math.floor(i / numItemsPerRow);
-        var items = isSingleRow ? numAllItems : numItemsPerRow;
-        var dx = col * legendItemWidth - items / 2 * legendItemWidth + _this.get('averageLegendLabelWidth') / 2;
+      transform: function(d, col) {
+        // Compute the assigned row and column for the current legend
+        var row = 0;
+        while (col >= numLegendItemsByRows[row]) {
+          col -= numLegendItemsByRows[row++];
+        }
+
+        // If first item in the row, set usedWidth as 0.
+        usedWidth = (col === 0) ? 0 : usedWidth;
```

Comment:
This should be an `if` statement. Really just across the board don't use the `? :` operator.